x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/e5_multilingual_callout/e5_multilingual_callout_logic.ts

```diff
@@ -0,0 +1,318 @@
+          'xpack.enterpriseSearch.content.indices.pipelines.e5MultilingualStartError.title',
+          {
+            defaultMessage: 'Error starting E5 Multilingual deployment',
+          }
+        ),
+        message: getErrorsFromHttpResponse(startError)[0],
+      }
+    : fetchError !== undefined
+    ? {
+        title: i18n.translate(
+          'xpack.enterpriseSearch.content.indices.pipelines.e5MultilingualFetchError.title',
+          {
+            defaultMessage: 'Error fetching E5 Multilingual model',
+          }
+        ),
+        message: getErrorsFromHttpResponse(fetchError)[0],
+      }
+    : null;
+};
+
+export const E5MultilingualCalloutLogic = kea<
+  MakeLogicType<E5MultilingualCalloutValues, E5MultilingualCalloutActions>
+>({
+  actions: {
+    clearE5MultilingualModelPollingId: true,
+    createE5MultilingualModelPollingTimeout: (duration) => ({ duration }),
+    setE5MultilingualModelPollingId: (pollTimeoutId: ReturnType<typeof setTimeout>) => ({
+      pollTimeoutId,
+    }),
+    startPollingE5MultilingualModel: true,
+    stopPollingE5MultilingualModel: true,
+    setE5MultilingualModelId: (e5MultilingualModelId) => ({ e5MultilingualModelId }),
+    createE5MultilingualModel: true,
+    fetchE5MultilingualModel: true,
+    startE5MultilingualModel: true,
+  },
+  connect: {
+    actions: [
+      CreateE5MultilingualModelApiLogic,
+      [
+        'makeRequest as createE5MultilingualModelMakeRequest',
+        'apiSuccess as createE5MultilingualModelSuccess',
+        'apiError as createE5MultilingualModelError',
+      ],
+      FetchE5MultilingualModelApiLogic,
+      [
+        'makeRequest as fetchE5MultilingualModelMakeRequest',
+        'apiSuccess as fetchE5MultilingualModelSuccess',
+        'apiError as fetchE5MultilingualModelError',
+      ],
+      StartE5MultilingualModelApiLogic,
+      [
+        'makeRequest as startE5MultilingualModelMakeRequest',
+        'apiSuccess as startE5MultilingualModelSuccess',
+        'apiError as startE5MultilingualModelError',
+      ],
+    ],
+    values: [
+      CreateE5MultilingualModelApiLogic,
+      [
+        'data as createdE5MultilingualModel',
+        'status as createE5MultilingualModelStatus',
+        'error as createE5MultilingualModelError',
+      ],
+      FetchE5MultilingualModelApiLogic,
+      ['data as e5MultilingualModel', 'error as fetchE5MultilingualModelError'],
+      StartE5MultilingualModelApiLogic,
+      ['status as startE5MultilingualModelStatus', 'error as startE5MultilingualModelError'],
+    ],
+  },
+  events: ({ actions, values }) => ({
+    afterMount: async () => {
+      actions.setE5MultilingualModelId('.multilingual-e5-small');
+      actions.fetchE5MultilingualModel();
+    },
+    beforeUnmount: () => {
+      if (values.e5MultilingualModelPollTimeoutId !== null) {
+        actions.stopPollingE5MultilingualModel();
+      }
+    },
+  }),
+  listeners: ({ actions, values }) => ({
+    createE5MultilingualModel: () =>
```

Comment:
@efegurkan 
It would be great if you can provide me with an example. Do you mean renaming **createE5MultilingualModel**?

Reply:
TLDR; I missed something yesterday, it can stay as is.

So currently, what `createE5MultilingualModel` listener does is simply to call `createE5MultilingualModelMakeRequest`. So this is an extra function call. I missed it yesterday that the call is actually using `modelId` from values therefore the outbound action that is used in the component avoids passing a parameter. 

For future reference;

in the connect section you can rename connected actions.
i.e. `'makeRequest as createE5MultilingualModel'` so you can use it in a component directly with `createE5MultilingualModal` without the need of creating a listener.
https://github.com/elastic/kibana/pull/171887/files#diff-84abd79feafce0616122640d64e6c92034d189413947aaf70f8358ca9cfd5e9eR146

For this case it is totally valid, please ignore.